Debian Buster image compatibility

amrsoll opened the pull request.

amrsoll committed:
Update numpy deps.

khaledsherkawi committed:
new material settings (#1162)

hungerpirat committed:
WorkingAreaHelper was not available in global scope, once assets were (#1161)

packed.

Co-authored-by: Andy Werner <Gallore@users.noreply.github.com>
Co-authored-by: khaledsherkawi <54568489+khaledsherkawi@users.noreply.github.com>

Gallore committed:
Bug fix: need to test if FORCE_REVIEW_SCREEN is declared (#1160)

* Bug fix: need to test if FORCE_REVIEW_SCREEN is declared

* Bug fix: Only ask new users.
Re-introducing the 'ask' setting and adding new "doNotAskAgain" setting
Also adding more metadata to usage handler.

* closing button fixed

Co-authored-by: khaledsherkawi <54568489+khaledsherkawi@users.noreply.github.com>

khaledsherkawi committed:
v0.8.2

khaledsherkawi committed:
v0.8.2.1: version bump

Gallore committed:
Minor corner calibration changes: logging and version added in pic_settings.yaml (#1164)

hungerpirat committed:
fixes #1169

Gallore committed:
Removes login credentials from frontend.log logfile (#1171)

Gallore committed:
Fixes failing translation of empty string (#1172)

* Fixes failing translation of empty string

* more typos in english material texts

khaledsherkawi committed:
Enable Design Store for Stable environment (#1175)

khaledsherkawi committed:
New material and settings added (#1174)

* new materials and settings added

* Polypropylene hint modified

amrsoll committed:
Fix user permissions for calibration and clean left over files. (#1173)

* Fix permissions and clean npz files.

* More cleaning before saving a picture.

Change comments.

* Restrict API to logged users or calibration_mode.

- Minor bugfix for removing unexpected files in debug dir.

Gallore committed:
Fix multiselect on Mac: support for metaKey (command) (#1177)

khaledsherkawi committed:
F whats new stable (#1140)

* whats new gcode deletion added for next stable release

* gcode deletion addition to the stable whats new wizard + modify the beta whats new wizard [better practice]

* removing unneccessary files

* skip gcode wizard if gcode deletion is already enabled + addition of demo next material list update

* remove gcode screen if it was already enabled

* whats new screen templates added

* whats new screens 2 and 4 modified

* deletion fix plus screen modification

* final edits

Gallore committed:
Final German translation (#1183)

khaledsherkawi committed:
v0.8.3

khaledsherkawi committed:
v0.8.3.1: version bump

khaledsherkawi committed:
v0.9.0

khaledsherkawi committed:
v0.9.0.1: version bump

hungerpirat committed:
bugfix. doubleclick in the corner of a working area moves laser now to (#1178)

this position.

hungerpirat committed:
B material height too large (#1179)

* bugfix. doubleclick in the corner of a working area moves laser now to
this position.

* bugfix. Limiting input thickness, limiting displayed material height.

* a bit more beautiful.

hungerpirat committed:
fixes firefox quicktext slider styling. (#1182)

amrsoll committed:
Use the new script names when changing AP and host-name.

Gallore committed:
Removed link to old forum (#1191)

Gallore committed:
Better logging on hw malfuncitons (#1185)

Gallore committed:
Converted string which get logged to unicode (#1198)

hungerpirat committed:
fixes #1194 - Duplicated QuickText: Can not be modified (#1199)

hungerpirat committed:
fixes #1192 Number input fields for design size change if a point number is entered with a comma. (#1200)

amrsoll committed:
rm leagcy bloat.

amrsoll committed:
Do not do all migrations from the dawn of time

amrsoll committed:
Remove auto os-healthcare.

amrsoll committed:
Make GRBL flasher py3 compatible.

amrsoll committed:
Skip rpiws281x updates.

amrsoll committed:
Ignore hostname if empty

Fixes #1205

amrsoll committed:
Fix software update plugin.

Commit:
Added try/catch for do_slice() method. Other wise exceptions there will never show.

amrsoll commented on octoprint_mrbeam/os_health_care.py:
```diff
@@ -12,6 +12,7 @@ def os_health_care(plugin):
 
 
 class OsHealthCare(object):
+    # FIXME : This should be part of the migration scripts
```
Could be  migrated to netconnectd

amrsoll replied:
i.e. if it fails to read the network file, assume a default version

andywerner commented on octoprint_mrbeam/util/device_info.py:
```diff
@@ -23,7 +23,6 @@ class DeviceInfo(object):
     KEY_HOSTNAME = "hostname"
     KEY_SERIAL = "serial"
     KEY_OCTOPI = "octopi"
-    KEY_IMAGE_CORRECTION_MARKERS = "image_correction_markers"
```
please keep this

amrsoll commented on setup.py:
```diff
@@ -36,7 +36,7 @@
 
 # Any additional requirements besides OctoPrint should be listed here
-plugin_requires = ["webcolors", "pillow", "lxml", "numpy==1.11.2", "picamera"]
+plugin_requires = ["webcolors", "pillow", "lxml", "numpy", "picamera"]
```
Find a way to use pip 9 and update with wheels

Gallore committed:
fixes issue with unicode in mrb_logger that prevented slicing. (#1216)

Also added try/catch in do_slice()

amrsoll committed:
Netconnectd doesn't update on older image.

amrsoll committed:
Fix datetime for py2 and iobeam socket.

amrsoll committed:
Track devpi.mr-beam.org

amrsoll committed:
Fix beamos date regex.

amrsoll committed:
update fixes for parsing

amrsoll committed:
Ignore package versions for now

amrsoll committed:
Fix undefined var.

amrsoll committed:
Compact / abstract software update info.

amrsoll committed:
Use `pip list` insteal of `pip freeze`

It shows the package version as the number instead of branches, path
etc...

amrsoll committed:
Fixes to software update

- added sudo for the required pip commands
- Added virtual_printer to the disabled plugins
- remove wrap_module decorator
- refactor self -> plugin
- multiple fixes (especially with the arguments)
- beamos_date discrimination for the related packages
- if package wasn't found, let us know that you can install the package
- logme importable from straight octoprint_mrbeam.util

amrsoll committed:
Fix logger levels

amrsoll committed:
Fix software update global pip.

andywerner commented on octoprint_mrbeam/__init__.py:
```diff
@@ -210,7 +210,9 @@ def initialize(self):
 
         # do os health care
-        os_health_care(self)
+        if False:
```
please change this.

andywerner commented on octoprint_mrbeam/__init__.py:
```diff
@@ -2680,34 +2682,41 @@ def getHostname(self):
+                    )
+                    # beamos_hostname only uses 4 characters to set the hostname as MrBeam-XXXX
+                    exec_cmd(
+                        "sudo /usr/bin/beamos_hostname {}".format(
```
Please check if `/usr/bin/beamos_hostname` exists and provide a fallback if not

andywerner commented on octoprint_mrbeam/iobeam/iobeam_handler.py:
```diff
@@ -75,7 +75,7 @@ class IoBeamHandler(object):
     #       echo "info" |  nc -U -w1 /var/run/mrbeam_iobeam.sock
 
-    SOCKET_FILE = "/var/run/mrbeam_iobeam.sock"
+    SOCKET_FILE = "/tmp/mrbeam_iobeam.sock"
```
make this backwards compatible

andywerner commented on octoprint_mrbeam/migrate.py:
```diff
@@ -817,3 +818,9 @@ def _set(path, _data, set_func, fullpath=None):
         settings().save()
         self._logger.info("Done.")
+
+    def track_devpi(self):
```
- needs to be called
- add try/catch
- add log output indication that this is about to happen

amrsoll committed:
Fix pip_command for the given package.

amrsoll committed:
Remove commented update listings.

amrsoll committed:
Keep legacy iobeam socket for this branch.

amrsoll committed:
Reinstate fixed numpy version.

First we need to find a way to point to a separate pip-compatible update
server Before using the migration scripts.

amrsoll committed:
OS Healthcare for the Jessie images.

amrsoll committed:
Make hostname change cross-img compatible.

amrsoll committed:
Add devpi tracking and version bump.

amrsoll committed:
Add logline for the devpi migration.

amrsoll committed:
Fix branch names for software update branches.

amrsoll committed:
Merge branch 'develop' into f_packaging_beamos

amrsoll committed:
Add the pink marker key back

amrsoll commented:
Current status: Parking this branch as we are waiting for an other beta release.

amrsoll committed:
Rm commnet

amrsoll committed:
Revert to numpy 1.16 and pip 18+ compat.

amrsoll committed:
Change default OctoPrint update info.

amrsoll committed:
Fix softwareupdate

- OctoPrint update mechanism -> `config.yaml`
- Fix displayVersion for certain packages.

amrsoll committed:
Allow for whitespace in /etc/mrbeam.

amrsoll committed:
Merge branch 'strip_etc_mrbeam' into f_packaging_beamos

amrsoll committed:
py3 compat

amrsoll committed:
Merge branch 'strip_etc_mrbeam' into f_packaging_beamos

Gallore committed:
Hard code login state in early stage (#1239)

See mrbeam/beamOS#42

amrsoll committed:
Pull 'f_packaging_beamos'

amrsoll committed:
Fix undefined variable in soft updt

amrsoll committed:
Bump version compared to develop branch.

amrsoll committed:
Merge remote-tracking branch 'origin/develop' into f_packaging_beamos

amrsoll committed:
Bump version again.

amrsoll committed:
Do not save debug img when path is none.

amrsoll committed:
Fix bug where .wait triggers "cam busy"

amrsoll committed:
Fix software update key.

amrsoll committed:
Add compatibility with camera plugin

amrsoll committed:
Fix timestamp formatting.

Commit:
Fix (potential) for login issue: checking if there is already a passive login in progress.

amrsoll committed:
Fixes

- json_serialisor can do atomic numpy arrays
- unicode support for some filepath
- fix loading previous result from npz file

amrsoll committed:
Lens calibration fixes

- Save calibration when min boars found in factory mode
- Fix a bunch of racing conditions etc..

amrsoll committed:
Stabilityfy lens calibration delete_picture.

amrsoll committed:
Fix branch names for updates w/ the newest BeamOS.

amrsoll commented:
Cosing in favor of #1265
